### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by RitobrotoMukherjee**
## Performed Tasks
* Used Capybara to write integration tests for each view in the project.
* Added tests for the user index page.
* Added tests for the user show page.
* Added tests for the user post index page.
* Added tests for the post-show page.

### CODE REVIEW

File: spec/views/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
```
Comment (gemmen29) [inline spec/views/post_index_spec.rb:1]: - [x] Kindly fix the failed tests.
image

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @RitobrotoMukherjee, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/post_index_spec.rb (hunk 1 above)
Comment (RitobrotoMukherjee) [reply spec/views/post_index_spec.rb:1]: Thank you for the update, I have fixed it now.

**Review by dasileker — APPROVED** [review]
### **Hi @RitobrotoMukherjee,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RitobrotoMukherjee** [comment]
@dasileker Thank you so much for approving my project. 🥳

**Pull request merged**